commit 2784e1a2dd79de0a13fe4cf75ce1d6ce8733d6a3
Author: Thomas Bruederli <bruederli@kolabsys.com>
Date:   Fri Nov 1 14:47:46 2013 +0100

    Fix replacement of alarm values

diff --git a/plugins/calendar/drivers/kolab/kolab_calendar.php b/plugins/calendar/drivers/kolab/kolab_calendar.php
index bd42f56..9eb70bd 100644
--- a/plugins/calendar/drivers/kolab/kolab_calendar.php
+++ b/plugins/calendar/drivers/kolab/kolab_calendar.php
@@ -581,7 +581,7 @@ class kolab_calendar
 
     // The web client only supports DISPLAY type of alarms
     if (!empty($record['alarms']))
-      $record['alarms'] = preg_replace('/:[A-Z]+$/', 'DISPLAY', $record['alarms']);
+      $record['alarms'] = preg_replace('/:[A-Z]+$/', ':DISPLAY', $record['alarms']);
 
     // remove empty recurrence array
     if (empty($record['recurrence']))
